app-near/src/server/external-api/api-su.ts

```diff
@@ -70,12 +68,63 @@ export const computeSus = async (
   }
 };
 
-export const verifyStep = async (surveyId: number) => {
-  const survey = await db.survey.findFirst({ where: { id: surveyId } });
-  if (!survey || survey.phase !== SurveyPhase.STEP_3_SU_EXPLORATION) {
+export const buildSuAssignmentRequest = async (
+  surveyId: number,
+  suAnswer: SuAnswer, // TODO: update with correct type
+): Promise<SuAssignementRequest> => {
+  const suDatas: SuData[] = await db.suData.findMany({
+    where: { surveyId },
+  });
+  return {
+    sus: suDatas.map((suData) => {
+      return { su: suData.su, barycenter: suData.barycenter as number[] };
+    }),
+    userData: convertToSuAnswerData(suAnswer),
+  };
+};
+
+export const assignSu = async (
+  payload: SuAssignementRequest,
+): Promise<AnswerAttributedSu> => {
+  try {
+    const snakeCaseAssignRequest = snakecaseKeys(
+      payload as unknown as Record<string, unknown>,
+    );
+
+    const response = await fetch(`${env.API_SU_URL}/api-su/assign`, {
+      method: "POST",
+      headers: {
+        "X-API-KEY": `${env.API_SU_KEY}`,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(snakeCaseAssignRequest),
+    });
+
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    const snakeCaseAssignResponse = await response.json();
+
+    if (!response.ok) {
+      throw new Error(
+        `Status=${response.status} Content=${snakeCaseAssignResponse}`,
+      );
+    }
+
+    const camelCaseAssignResponse = camelcaseKeys(snakeCaseAssignResponse, {
+      deep: true,
+    }) as AnswerAttributedSu;
+
+    return camelCaseAssignResponse;
+  } catch (error) {
+    console.error("Error calling API SU compute:", error);
+    if (error instanceof TypeError) {
+      throw new TRPCError({
+        code: "INTERNAL_SERVER_ERROR",
+        message: "Network error occurred while calling the API.",
```

Comment:
```suggestion
        message: "Network error occurred while calling the su API.",
```
Ca sera un message renvoyé au client. Si c'est affiché dans le toaster de l'app, on peut éventuellement dire de quelle API on parle